### ISSUE TITLE
Debugging a Source Generator when added to an xUnit testing project does not work

### ISSUE THREAD

**Issue by JoanComasFdz** [dotnet/roslyn-sdk#972]

**Comment by JoanComasFdz** [comment]
I tried adding the generated files to the source control as explained [here](https://andrewlock.net/creating-a-source-generator-part-6-saving-source-generator-output-in-source-control/), by adding the following lines to the xUnit project .csproj file:

```
<PropertyGroup>
    <EmitCompilerGeneratedFiles>true</EmitCompilerGeneratedFiles>
    <CompilerGeneratedFilesOutputPath>Generated</CompilerGeneratedFilesOutputPath>
</PropertyGroup>
```

And the folder and files are **not** being generated. So I guess the issue is not really debugging, but a lack of support for xUnit?

I am trying to make this work in this repo, maybe it helps identifying the issue: 
https://github.com/JoanComasFdz/dotnet-scenario-unittesting

**Comment by JoanComasFdz** [comment]
Thanks for the feedback @sharwell. 

Just to make sure we are talking about the same thing:

* I am **not** trying to create and debug a unit test for my source generator.
* I am trying to create and debug a source generator to be added in xUnit test projects.

I mention this because of your sentences "add a source generator test to the project" and "debug the test via Test Explorer".

Does the setup you propose allow to debug a source generator referenced in a xUnit project?

**Comment by sharwell** [comment]
Yes, but debugging that source generator would involve creating a second test project specifically for testing (and not using) that source generator.
